platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationEntry.java

```diff
@@ -0,0 +1,108 @@
+     *
+     * @return the responsible migratable service id
+     */
+    public String getId();
+
+    /**
+     * Gets name for this entry. The name is standardized with slashes (i.e /).
+     *
+     * @return the name for this entry
+     */
+    public String getName();
+
+    /**
+     * Gets a {@link Path} for this entry.
+     * <p>
+     * <i>Note:</i> Absolute paths that are under ${ddf.home} are automatically relativized.
+     *
+     * @return a path for this entry
+     */
+    public Path getPath();
+
+    /**
+     * Gets the last modification time of the entry.
+     *
+     * @return the last modification time of the entry in milliseconds since the epoch,
+     * or -1 if not specified
+     */
+    public long getLastModifiedTime();
+
+    /**
+     * Stores this entry's content in the export during an export migration operation or underneath
+     * the distribution root directory during an import migration operation based on this entry's
+     * path which can include sub-directories.
+     * <p>
+     * During an import migration operation, this entry's sub-directories (if any) will be created if
+     * they don't already exist. The destination file will be overwritten if it already exists.
+     * <p>
+     * All errors and warnings are automatically recorded with the associated migration report.
+     *
+     * @return <code>true</code> if no errors were recorded as a result of processing this command;
+     * <code>false</code> otherwise
+     * @throws MigrationException if a failure that prevents the operation from continue occurred
+     */
+    public default boolean store() {
+        return store(true);
+    }
+
+    /**
+     * Stores this entry's content in the export during an export migration operation or underneath
+     * the distribution root directory during an import migration operation based on this entry's
+     * path which can include sub-directories.
+     * <p>
+     * During an import migration operation, this entry's sub-directories (if any) will be created if
+     * they don't already exist. The destination file will be overwritten if it already exists.
+     * <p>
+     * All errors and warnings are automatically recorded with the associated migration report.
+     *
+     * @param required <code>true</code> if the file is required to exist on disk during an export
+     *                 migration operation or in the export during an import migration operation and
+     *                 if it doesn't an error should be recorded; <code>false</code> if the file is
+     *                 optional and may not be present or exported in which case calling this method
+     *                 will do nothing
+     * @return <code>true</code> if no errors were recorded as a result of processing this command;
+     * <code>false</code> otherwise
+     * @throws MigrationException if a failure that prevents the operation from continue occurred
+     */
+    public boolean store(boolean required);
```

Comment:
❓ Should we call the method `storeRequired()` (or something similar) and get rid of the `boolean` flag?